db/version_set.cc

```diff
@@ -3745,12 +3748,27 @@ Status VersionSet::ProcessManifestWrites(
     // No group commits for column family add or drop
     LogAndApplyCFHelper(first_writer.edit_list.front());
     batch_edits.push_back(first_writer.edit_list.front());
+  } else if (first_writer.edit_list.front()->IsWalManipulation()) {
+    // WAL addition/deletion won't appear together with other kinds of edits,
```

Comment:
Why cannot one writer perform MANIFEST write for another writer's WAL manipulations?

Reply:
There is no reason that WAL edits cannot be grouped.

But since WAL edits are not treated differently from other kinds of edits (such as not applied to versions, and needs to update `wals_`), handling them specially like column family edits makes the logic easier to understand.

Reply:
I feel that additional logic to (unnecessarily) separate the processing of wal-related edits make the `ProcessManifestWrites` more complicated than before. Can we do something like
```
for (const auto& edit : edits) {
  if (wal-related) {
    LogAndApplyWalManipulation();
  } else {
    LogAndApplyHelper();
  }
}
```

Reply:
the tricky part is to handle `version` related logic differently for WAL edits, there is no need to update versions for WAL edits.

Reply:
updated the logic, PTAL